### PR TITLE
[hmac_smoketest] change endianness of expectation

### PR DESCRIPTION
**Pull request by drewmacrae**
https://github.com/lowRISC/opentitan/pull/9553 has swapped the
endianness bit to correct it's behavior so the smoke test for the hmac
needs to have a corrected expectation to pass.

Signed-off-by: Drew Macrae <drewmacrae@google.com>

### CODE REVIEW

**Comment by drewmacrae** [comment]
I think I'm reading the endianness right for the sha256 correctly "Every one suspects himself of at least one of the cardinal virtues, and this is mine: I am one of the few honest people that I have ever known" should be 3dc296dc68e236af71ff68cbe2762fe99d37a8b845c76d42f7cff519d6c6c94e if little endian as is done in the smoketest in this commit.

**Review by jadephilipoom — APPROVED** [review]
Yes, this matches my understanding. In particular, when I compute the SHA-256 digest for the message, I get `0x3dc296dc68e236af71ff68cbe2762fe99d37a8b845c76d42f7cff519d6c6c94e` (expressed in big-endian notation). This matches the new expected digest when split into words.

It's slightly confusing that the expected digest looks like big-endian form when all the options say "little", so let me add a quick note here for anyone reading this in the future. This is because a) C expresses individual integers in big-endian even on a little-endian processor, so these are essentially reversed when stored, and b) the HMAC device produces big-endian _word_ order no matter what the endianness options are; the `digest_endianness` affects only the order in which the bytes _within a word_ are read. Similarly, the `message_endianness` only affects multi-byte writes, and just decides how to order those bytes; it should always be "little" for a little-endian processor, even for dealing with big-endian buffers, because it might be convenient to change 4 byte writes to a word write when transferring the buffer.

tl;dr the big/little options for HMAC have everything to do with processor/single-word endianness, and nothing to do with array endianness.

**Comment by alphan** [comment]
> tl;dr the big/little options for HMAC have everything to do with processor/single-word endianness, and nothing to do with array endianness.

That's my understanding as well. Just wanted to add that in `sw/device/silicon_creator/lib/drivers/hmac.c` we read the words in reverse, i.e. DIGEST_7 to DIGEST_0, to preserve the numerical value. [HMAC hwip spec](https://docs.opentitan.org/hw/ip/hmac/doc/) also says:

> Order of the digest is: digest[255:0] = {DIGEST0, DIGEST1, DIGEST2, ... , DIGEST7};

@jadephilipoom would you mind taking a look at `sw/device/silicon_creator/lib/drivers/hmac.c` to see if the configuration there matches your understanding?

**Review by alphan — APPROVED** [review]

**Comment by alphan** [comment]
I requested a re-run of CW310 bitstream step just in case but I wouldn't be surprised if it fails again. First run failed with the following error:

```
INFO: [Designcheck 1-1] Checking design
WARNING: [Timing 38-172] LUT was found on clock network. Both rising/falling clock edges are propagated from LUT output pin. Use set_clock_sense to specify if that clock should be inverted or not. e.g. set_clock_sense -positive/-negative top_earlgrey/u_clkmgr_aon/u_no_scan_io_div2_div/gen_div2.u_div2/gen_fpga_buf.bufg_i_i_1__0/O
get_timing_paths: Time (s): cpu = 00:02:49 ; elapsed = 00:00:29 . Memory (MB): peak = 6477.715 ; gain = 0.000 ; free physical = 29312 ; free virtual = 53187
INFO: [Designcheck 1-2] Slack is -0.277 ns.
ERROR: [Designcheck 1-3] Timing failed. Slack is -0.277 ns.
ERROR: [Common 17-39] 'send_msg_id' failed due to earlier errors.
```

@drewmacrae can you check with hw folks to see if this is a known issue?

**Comment by jadephilipoom** [comment]
> @jadephilipoom would you mind taking a look at sw/device/silicon_creator/lib/drivers/hmac.c to see if the configuration there matches your understanding?

Yes, it matches! I actually checked there too when I made #9553, but took another look now and it still looks good.

> @drewmacrae can you check with hw folks to see if this is a known issue?

Looks like it might be the issue #9986 is trying to fix?

**Comment by drewmacrae** [comment]
Rebased to head and pushed to rerun CI
